### PR TITLE
feat(agent): no-code update record should handle errors correctly

### PR DESCRIPTION
**Pull request by Thenkei**
### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

**Comment by realSpok** [comment]
Nice !
image

File: packages/agent/src/services/model-customizations/actions/update-record/execute-update-record.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      return { ...acc, [fieldToUpdate.fieldName]: fieldToUpdate.value };
+    }, {});
+
+    await context.collection.update(context.filter, fields);
+  } catch (error) {
+    if (error instanceof BusinessError) {
+      return {
+        type: 'Error',
+        message:
+          `The no-code action ${action.name} cannot be triggered due to misconfiguration.` +
```
Comment (realSpok) [inline packages/agent/src/services/model-customizations/actions/update-record/execute-update-record.ts:26]:
```suggestion
          `The no-code action ${action.name} cannot be triggered due to a misconfiguration.`
```
It is up for debate but we could remove the 'no-code'. From the user's perspective there are no difference between actions.

File: packages/agent/src/services/model-customizations/actions/update-record/execute-update-record.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+          ` Please contact your administrator.\n(${error.name}: ${error.message})`,
```
Comment (realSpok) [inline packages/agent/src/services/model-customizations/actions/update-record/execute-update-record.ts:27]: The '\n' line break doesn't seem to be working. Do we need a <br/> ? or something else ?

File: packages/agent/src/services/model-customizations/actions/update-record/execute-update-record.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      };
+    }
+
+    return {
+      type: 'Error',
+      message:
+        `The no-code action ${action.name} cannot be triggered due to misconfiguration.` +
```
Comment (realSpok) [inline packages/agent/src/services/model-customizations/actions/update-record/execute-update-record.ts:34]: Same as above

Comment (realSpok) [reply packages/agent/src/services/model-customizations/actions/update-record/execute-update-record.ts:34]: We could maybe factorize those 2 messages into a function

File: packages/agent/src/services/model-customizations/actions/update-record/execute-update-record.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  context: ActionContext<S>,
+): Promise<ActionResult> {
+  try {
+    const {
+      configuration: {
+        configuration: { fields: fieldsToUpdate },
+      },
+    } = action;
+
+    const fields = fieldsToUpdate.reduce((acc, fieldToUpdate) => {
```
Comment (realSpok) [inline packages/agent/src/services/model-customizations/actions/update-record/execute-update-record.ts:16]: I am not sure what `acc` stands for …

**Review by realSpok — CHANGES_REQUESTED** [review]
- ❗  Code review
  - ✅  Use pure functions when possible
  - ✅  Performance concerns
  - ✅  Security concerns
  - ✅  The PR tackle one subject only
- ✅ Automatic tests
  - ✅  Unit tests
  - ⤬  Integration tests
- ✅ Manual tests
  -  ✅ Test the functionality
  -  ✅ Test error cases
- ✅  PR title:
- ✅  Destination branch
- ✅  PR linked to the clickup task

File: packages/agent/src/services/model-customizations/actions/update-record/execute-update-record.ts (hunk 1 above)
Comment (realSpok) [reply packages/agent/src/services/model-customizations/actions/update-record/execute-update-record.ts:26]: Even if the 'no-code' is a marketing term, I would say it is for the admins, not the users

File: packages/agent/src/services/model-customizations/actions/update-record/execute-update-record.ts (hunk 4 above)
Comment (Thenkei) [reply packages/agent/src/services/model-customizations/actions/update-record/execute-update-record.ts:16]: accumulator 😥

Comment (realSpok) [reply packages/agent/src/services/model-customizations/actions/update-record/execute-update-record.ts:16]: ok. I guess I am also not very comfortable with reduce :)
But I get it now. You can leave it like that or write it as `accumulator`, as you prefer

File: packages/agent/src/services/model-customizations/actions/update-record/execute-update-record.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+): Promise<ActionResult> {
+  try {
+    const {
+      configuration: {
+        configuration: { fields: fieldsToUpdate },
+      },
+    } = action;
+
+    const fields = fieldsToUpdate.reduce((fieldsAccumulator, fieldToUpdate) => {
+      return { ...fieldsAccumulator, [fieldToUpdate.fieldName]: fieldToUpdate.value };
```
Comment (realSpok) [inline packages/agent/src/services/model-customizations/actions/update-record/execute-update-record.ts:21]: `fieldsAccumulator` -> very straigthforward !

**Pull request merged**